docs/source/user/cuda_ffi.rst

```diff
@@ -160,6 +160,19 @@ CUDA C/C++ source code will be compiled with the `NVIDIA Runtime Compiler
 kernel as either PTX or LTOIR, depending on whether LTO is enabled. Other files
 will be passed directly to the CUDA Linker.
 
+For cumodules linked to a linkable code object, additional setup
+and teardown callbacks are available. Setup functions are invoked once for
+every new module loaded. Teardown functions are invoked before the module is
+garbage collected by python. These callbacks are defined as follows:
```

Comment:
I think the module is getting unloaded during the GC process or when the refcount has already dropped to zero outside of a reference cycle, so maybe the following is a closer representation of what happens:

```suggestion
every new module loaded. Teardown functions are invoked just prior to
module unloading. These callbacks are defined as follows:
```